src/H5Dlayout.c

```diff
@@ -271,191 +271,6 @@ H5D__layout_meta_size(const H5F_t *f, const H5O_layout_t *layout, bool include_c
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5D__layout_meta_size() */
 
-/*-------------------------------------------------------------------------
- * Function:    H5D__layout_set_version
- *
- * Purpose:     Set the version to encode a layout with.
- *
- * Return:      Non-negative on success/Negative on failure
- *-------------------------------------------------------------------------
- */
-herr_t
-H5D__layout_set_version(H5F_t *f, H5O_layout_t *layout)
```

Comment:
Functionality moved to the construct callbacks and H5D__create (for version bounds check)